lua/lua/usip_generator.lua

```diff
@@ -25,98 +25,118 @@ function UsipGenerator:_init(tLog)
 
         -- add key index
-        tUsipFileHandle:write(tChunkContent['key_idx'])
+        strUsipData = strUsipData .. tChunkContent['key_idx']
 
         -- add patched size
-        tUsipFileHandle:write(tChunkContent['patched_size'])
+        strUsipData = strUsipData .. tChunkContent['patched_size']
 
         if tChunkContent['key_idx_int'] ~= 255 then
-            tUsipFileHandle:write(tChunkContent['uuid'])
-            tUsipFileHandle:write(tChunkContent['anchor_0'])
-            tUsipFileHandle:write(tChunkContent['anchor_1'])
-            tUsipFileHandle:write(tChunkContent['anchor_2'])
-            tUsipFileHandle:write(tChunkContent['anchor_3'])
-            tUsipFileHandle:write(tChunkContent['uuid_mask'])
-            tUsipFileHandle:write(tChunkContent['anchor_mask_0'])
-            tUsipFileHandle:write(tChunkContent['anchor_mask_1'])
-            tUsipFileHandle:write(tChunkContent['anchor_mask_2'])
-            tUsipFileHandle:write(tChunkContent['anchor_mask_3'])
-            tUsipFileHandle:write(tChunkContent['padded_key'])
+            strUsipData = strUsipData .. tChunkContent['uuid']
+            strUsipData = strUsipData .. tChunkContent['anchor_0']
+            strUsipData = strUsipData .. tChunkContent['anchor_1']
+            strUsipData = strUsipData .. tChunkContent['anchor_2']
+            strUsipData = strUsipData .. tChunkContent['anchor_3']
+            strUsipData = strUsipData .. tChunkContent['uuid_mask']
+            strUsipData = strUsipData .. tChunkContent['anchor_mask_0']
+            strUsipData = strUsipData .. tChunkContent['anchor_mask_1']
+            strUsipData = strUsipData .. tChunkContent['anchor_mask_2']
+            strUsipData = strUsipData .. tChunkContent['anchor_mask_3']
+            strUsipData = strUsipData .. tChunkContent['padded_key']
         end
 
         -- add data
         for iDataIdx=0, tChunkContent['ulDataCount'] do
-            tUsipFileHandle:write(tChunkContent['data'][iDataIdx]['offset'])
-            tUsipFileHandle:write(tChunkContent['data'][iDataIdx]['size'])
-            tUsipFileHandle:write(tChunkContent['data'][iDataIdx]['patched_data'])
+            strUsipData = strUsipData .. tChunkContent['data'][iDataIdx]['offset']
+            strUsipData = strUsipData .. tChunkContent['data'][iDataIdx]['size']
+            strUsipData = strUsipData .. tChunkContent['data'][iDataIdx]['patched_data']
         end
 
         -- add padding
-        tUsipFileHandle:write(tChunkContent['padding'])
+        strUsipData = strUsipData .. tChunkContent['padding']
 
         --add signature
-        tUsipFileHandle:write(tChunkContent['signature'])
+        strUsipData = strUsipData .. tChunkContent['signature']
         -- add 4 zeros
-        tUsipFileHandle:write(string.char(0x00, 0x00, 0x00, 0x00))
+        strUsipData = strUsipData .. string.char(0x00, 0x00, 0x00, 0x00)
+
+        table.insert(aDataList, strUsipData)
```

Comment:
suggestion: use just one list atData, for each file, add an entry 
table.insert(atData, {content = strUsipData, path="single_usip".. iIdx})
This makes iterating easier, too.